src/main/java/core/basesyntax/Storage.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public void setFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void addFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void reduceFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
+    }
+
+    public Map<String, Integer> getAllFruits() {
```

Comment:
This method is returning a direct reference to the `fruits` map. This violates the principle of data hiding and encapsulation. You should return a copy of the map or an unmodifiable map.

Reply:
changes have been made